x/mongo/driver/operation/count.go

```diff
@@ -102,16 +127,63 @@ func (c *Count) Execute(ctx context.Context) error {
 		ServerAPI:         c.serverAPI,
 	}.Execute(ctx, nil)
 
+	// Swallow error if NamespaceNotFound(26) is returned from aggregate on non-existent namespace
+	if err != nil {
+		dErr, ok := err.(driver.Error)
+		if ok && dErr.Code == 26 {
+			err = nil
+		}
+	}
+	return err
 }
 
 func (c *Count) command(dst []byte, desc description.SelectedServer) ([]byte, error) {
-	dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+	// If wire version < 12 (4.9.0), use count command
+	if desc.WireVersion.Max < 12 {
+		dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+		if c.maxTimeMS != nil {
+			dst = bsoncore.AppendInt64Element(dst, "maxTimeMS", *c.maxTimeMS)
+		}
+		if c.query != nil {
+			dst = bsoncore.AppendDocumentElement(dst, "query", c.query)
+		}
+		return dst, nil
+	}
+	// If wire version >= 12 (4.9.0), use aggregate with $collStats
+	header := bsoncore.Value{Type: bsontype.String, Data: bsoncore.AppendString(nil, c.collection)}
```

Comment:
Looks like this is copied over from `aggregate.go`. In that operation, this code is required because the `aggregate` command can be run against a collection or database. For the latter, the command looks like `{aggregate: 1}`.

Here, however, we know we're running a `EstimatedDocumentCount` operation, which can only run against a collection so you can just do

```
dst = bsoncore.AppendStringElement(dst, "aggregate", c.collection)
```